streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -85,6 +85,21 @@ public void initializeMetadata() {}
     public boolean initializeStateStores() {
         log.trace("Initializing state stores");
         registerStateStores();
+
+        if (eosEnabled) {
```

Comment:
I have always been, and continue to be, mystified about why we're 100% a-ok with having corrupted state stores in ALOS mode. I understand that there are some kinds of corruption that are unavoidable without some extra work in the stores, but this check is just explicitly ignoring detectable corruption, which seems a bit extreme. Can we remove this conditional? The rest of these checks seem to apply equally well under all execution semantic modes.

Reply:
That is the tradeoff people buy in when using at-least-once. As long as state store recovery is expensive, it's not worth to change it IMHO. Also note, even if we wipe out the store for non-eos, the corrupted record might have been written into the changelog topic already, and thus, we would recover the corrupted state again, not gaining anything.